etna/records/views/records.py

```diff
@@ -76,10 +83,28 @@ def record_detail_view(request, iaid):
     # if page.is_digitised:
     #     image = Image.search.filter(rid=page.media_reference_id).first()
 
+    # Back to search - default url
+    back_to_search_url = reverse("search-featured")
+
+    # Back to search button - update url when timestamp is not expired
+
+    if back_to_search_url_timestamp := request.session.get(
+        "back_to_search_url_timestamp"
+    ):
+        back_to_search_url_timestamp = datetime.datetime.fromisoformat(
+            back_to_search_url_timestamp
+        )
+        back_to_search_url_timestamp_delta = (
+            back_to_search_url_timestamp + SEARCH_URL_RETAIN_DELTA
+        )
+        if timezone.now() <= back_to_search_url_timestamp_delta:
```

Comment:
Non-blocking: There's no need to define a variable for everything, especially when variable names are so lengthy. You could just calculate and test in a single line here:

```suggestion
        if timezone.now() <= (back_to_search_url_timestamp + SEARCH_URL_RETAIN_DELTA):
```

Reply:
I did this somewhere else in an assignment and missed it here - `session["back_to_search_url_timestamp"] = (self.back_to_search_url_timestamp - SEARCH_URL_RETAIN_DELTA).isoformat()` - Fixed